pybliometrics/sciencedirect/object_retrieval.py

```diff
@@ -0,0 +1,65 @@
+"""Module to retrieve a specific object of a document."""
+from io import BytesIO
+from typing import Optional, Union
+
+from pybliometrics.sciencedirect import ArticleRetrieval
+from pybliometrics.superclasses import Retrieval
+from pybliometrics.utils import check_parameter_value, detect_id_type
+
+
+class ObjectRetrieval(Retrieval):
+    @property
+    def object(self) -> BytesIO:
+        """The object retrieved."""
+        return BytesIO(self._object)
+
+    def __init__(self,
+                 identifier: Union[int, str],
+                 filename: str,
+                 id_type: Optional[str] = None,
+                 refresh: Union[bool, int] = False,
+                 **kwds: str
+                 ):
+        """Class to retrieve a specific object of a document by its filename.
+
+        :param identifier: The indentifier of the document.
+        :param filename: Filename of the object to be retrieved. To get a list of all available
+            objects of a document (and its corresponding filename) use the class `ObjectMetadata`.
+        :param id_type: Document identifier. Allowed values: `doi`, `pii`, `scopus_id`, 
+            `pubmed_id`, `eid`.
+        :param refresh: Whether to refresh the cached file if it exists. Default: False.
+
+        Example:
+            >>> from PIL import Image
+            >>> from pybliometrics.sciencedirect import init, ObjectMetadata, ObjectRetrieval
+            >>> init()
+            >>> # Get all objects and its filenames
+            >>> om = ObjectMetadata('10.1016/j.rcim.2020.102086')
+            >>> all_filenames = [f.filename for f in om.results]
+            >>> # Get 6th object: gr7.jpg
+            >>> obj_ret = ObjectRetrieval('10.1016/j.rcim.2020.102086', all_filenames[5])
+            >>> # Display using PIL
+            >>> img = Image.open(obj_ret.object)
+            >>> img.show()
```

Comment:
Normally we have examples on a dedicate reference page. We can change that eventually, but for this class we should stick to the norm that we have.